lib/TPP/Conversion/ConvertVectorToXsmm/ConvertVectorToXsmm.cpp

```diff
@@ -0,0 +1,235 @@
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Conversion/ConvertVectorToXsmm/ConvertVectorToXsmm.h"
+#include "TPP/Dialect/Xsmm/XsmmUtils.h"
+#include "TPP/Transforms/Utils/BuilderUtils.h"
+#include "TPP/Transforms/Utils/VNNIUtils.h"
+#include "TPP/Transforms/Utils/ValueUtils.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/PDL/IR/PDL.h"
+#include "mlir/Dialect/PDLInterp/IR/PDLInterp.h"
+#include "mlir/Dialect/Transform/PDLExtension/PDLExtensionOps.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+#include "mlir/IR/BuiltinDialect.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/Support/Debug.h"
+using namespace mlir;
+using namespace mlir::vector;
+using namespace mlir::linalg;
+using namespace mlir::func;
+using namespace mlir::xsmm;
+
+#include "ConvertVectorToXsmmPDLLPatterns.h.inc"
+
+#define DEBUG_TYPE "convert-vector-to-xsmm"
+
+pair<Operation *, Operation *>
+getTransposeOperation(PatternRewriter &rewriter, xsmm::UnaryInfo &unaryInfo,
+                      Type outputType, xsmm::UnaryKind unaryKind,
+                      Operation *input, Operation *output,
+                      Operation *transposeOp) {
+  std::string dispatchName = "xsmm_unary_dispatch";
+  std::string invokeName = "xsmm_unary_invoke";
+  Location loc = transposeOp->getLoc();
+
+  ModuleOp module = transposeOp->getParentOfType<ModuleOp>();
+  auto dtype =
+      xsmm::utils::getDataType(rewriter, transposeOp->getOperand(0).getType());
+
+  auto functionOp = transposeOp->getParentOfType<func::FuncOp>();
+  SmallVector<Value, 10> dispatchOperands;
+  OpBuilder::InsertionGuard guard(rewriter);
+  rewriter.setInsertionPoint(&*functionOp.getBody().op_begin());
+  xsmm::UnaryKindAttr kind =
+      xsmm::UnaryKindAttr::get(rewriter.getContext(), unaryKind);
+  dispatchOperands.push_back(
+      getConstInt(rewriter, dyn_cast<IntegerAttr>(kind).getInt(), 64));
+
+  dispatchOperands.push_back(
+      getConstInt(rewriter, dyn_cast<DataTypeAttr>(dtype).getInt(), 64));
+
+  DenseI64ArrayAttr dims = DenseI64ArrayAttr::get(
+      rewriter.getContext(), ArrayRef<int64_t>{unaryInfo.m, unaryInfo.n,
+                                               unaryInfo.ldi, unaryInfo.ldo});
+  for (auto idx = 0; idx < dims.size(); idx++) {
+    dispatchOperands.push_back(getConstInt(rewriter, dims[idx], 64));
+  }
+  IntegerAttr unaryFlags = dyn_cast<IntegerAttr>(
+      xsmm::UnaryFlagsAttr::get(rewriter.getContext(), xsmm::UnaryFlags::NONE));
+  dispatchOperands.push_back(getConstInt(rewriter, unaryFlags.getInt(), 64));
+
+  auto dispatched = xsmm::utils::buildXsmmCall(
```

Comment:
As discussed in the definition, `buildXsmmCall` is ambiguous. You need to make it clear which of them you're changing it (dispatch, invoke).

If you have two calls (or a type argument), then you don't even need to change the insertion point in this function. You can set it for _dispatch_ and the insertion guard will make it sure that when you build the `invoke`, it will be back in the right place.